amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java

```diff
@@ -76,47 +58,58 @@ public void testMultiStreamDeserializationFail() {
         }
     }
 
+    /**
+     * Test patterns that <b>should not</b> match a valid streamARN regex.
+     */
     @Test
-    public void testInstanceFromArn() {
-        final Arn arn = toArn(KINESIS_REGION);
-        final StreamIdentifier single = StreamIdentifier.singleStreamInstance(arn.toString());
-        final StreamIdentifier multi = StreamIdentifier.multiStreamInstance(arn.toString());
-
-        assertEquals(single, multi);
-        assertEquals(Optional.empty(), single.streamCreationEpochOptional());
-        assertActualStreamIdentifierExpected(arn, single);
-    }
-
-    @Test(expected = IllegalArgumentException.class)
-    public void testInstanceWithoutEpochOrArn() {
-        mockStatic(StreamARNUtil.class);
-        when(getStreamARN(STREAM_NAME, KINESIS_REGION, TEST_ACCOUNT_ID))
-                .thenReturn(Optional.empty());
-
-        try {
-            StreamIdentifier.singleStreamInstance(DEFAULT_ARN.toString());
-        } finally {
-            verifyStatic(StreamARNUtil.class);
-            getStreamARN(STREAM_NAME, KINESIS_REGION, TEST_ACCOUNT_ID);
+    public void testMultiStreamArnFail() {
+        for (final String pattern : Arrays.asList(
+                "arn:aws:kinesis::123456789012:stream/stream-name", // missing region
+                "arn:aws:kinesis:region::stream/stream-name", // missing account id
+                "arn:aws:kinesis:region:123456789:stream/stream-name", // account id not 12 digits
+                "arn:aws:kinesis:region:123456789abc:stream/stream-name", // 12char alphanumeric account id
+                "arn:aws:kinesis:region:123456789012:stream/" // missing stream-name
+        )) {
+            try {
+                StreamIdentifier.multiStreamInstance(Arn.fromString(pattern), EPOCH);
```

Comment:
Please move `Arn.fromString(...)` before the `try`. We should not allow an IAE thrown from Arn -- if it even does, now or in the future -- to create false assurance for this test.

Reply:
This was indeed creating false assurance for this test. Updated test to focus on Arns that can be successfully constructed with `Arn.Builder()` but are not valid Kinesis stream Arns.